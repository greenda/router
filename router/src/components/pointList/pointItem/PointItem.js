import React from 'react'
import { Draggable } from 'react-beautiful-dnd'
import './PointItem.scss'

const getItemStyle = (isDragging, draggableStyle) => ({
    background: isDragging ? 'lightgreen' : 'grey',  
    ...draggableStyle
});

export function PointItem({point, index}) {
    const { name } = point
    
    return (
        <Draggable key={point.id} draggableId={point.id} index={index}>
            {(provided, snapshot) => (
            <div 
                ref={provided.innerRef}
                {...provided.draggableProps}
                {...provided.dragHandleProps}
                style={getItemStyle(
                snapshot.isDragging,
                provided.draggableProps.style
                )}
                className='point-item'
            >
                <div data-testid="task-content" className="point-item__name">{name}</div>
                <div className="point-item__remove-icon">x</div>     
            </div>
            )}
        </Draggable>
    )
}

export default PointItem