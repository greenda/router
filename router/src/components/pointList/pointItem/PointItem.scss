@import '../../../common/style/variables.scss';

.point-item {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    padding: $offset__small;
    
    &__remove-icon {
        font-weight: bold;
        padding: 4px;
        border-radius: 4px;
        height: 16px;
        display: flex;
        align-items: flex-end;
        width: 16px;
        justify-content: center;
    }
}