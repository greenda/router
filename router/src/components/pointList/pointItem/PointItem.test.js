import React from 'react';
import { PointItem } from './PointItem'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'

const mockPoint = {
	id: 1,
	coordinates: [-41.28, 174.69],
	name: 'first point'
}

it('render correctly point item component', () => {
	const tree = renderer.create(
		<PointItem point={mockPoint} />
	  ).toJSON();
	expect(tree).toMatchSnapshot();
})

it('call remove function', () => {
	const removeFunction = (pointId) => {
		expect(pointId).toEqual(mockPoint.id)		
	}

	const pointItem = mount(
		<PointItem point={mockPoint} removePoint={removeFunction}/>
	)
	pointItem.find('.point-item__remove-icon').simulate('click')
})
