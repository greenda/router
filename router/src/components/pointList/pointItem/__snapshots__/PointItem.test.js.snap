// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly point item component 1`] = `
<div
  className="point-item"
>
  <div
    className="point-item__name"
    data-testid="task-content"
  >
    first point
  </div>
  <div
    className="point-item__remove-icon"
    onClick={[Function]}
  >
    x
  </div>
</div>
`;
